src/software/ai/navigator/path_planner/theta_star_path_planner.cpp

```diff
@@ -172,95 +166,95 @@ bool ThetaStarPathPlanner::updateVertex(Coordinate current_coord, Coordinate new
 
 // top level function
 std::optional<Path> ThetaStarPathPlanner::findPath(
-    const Point &start, const Point &destination, const Rectangle &navigable_area,
+    const Point &start, const Point &end, const Rectangle &navigable_area,
     const std::vector<ObstaclePtr> &obstacles)
 {
     resetAndInitializeMemberVariables(navigable_area, obstacles);
 
-    Point closest_destination = findClosestFreePoint(destination);
-    Coordinate src_coord      = pointToCoordinate(start);
-    Coordinate dest_coord     = pointToCoordinate(closest_destination);
+    Point closest_end      = findClosestFreePoint(end);
+    Coordinate start_coord = convertPointToCoord(start);
+    Coordinate end_coord   = convertPointToCoord(closest_end);
 
-    bool isInvalidOrBlocked = checkForInvalidOrBlockedCases(src_coord, dest_coord);
+    bool isInvalidOrBlocked = checkForInvalidOrBlockedCases(start_coord, end_coord);
     if (isInvalidOrBlocked)
     {
         return std::nullopt;
     }
 
-    if (isStartToDestinationWithinThreshold(start, destination))
+    if (isStartToEndWithinThreshold(start, end))
     {
-        return Path(std::vector<Point>({start, destination}));
+        return Path(std::vector<Point>({start, end}));
     }
 
-    if (isStartToClosestDestinationWithinThreshold(start, closest_destination))
+    if (isStartToClosestEndWithinThreshold(start, closest_end))
     {
-        return Path(std::vector<Point>({start, closest_destination}));
+        return Path(std::vector<Point>({start, closest_end}));
     }
 
-    // Initialising the parameters of the starting node
-    cell_heuristics[src_coord.row()][src_coord.col()].update(src_coord, 0.0, 0.0);
-    open_list.insert(src_coord);
+    // Initialising the parameters of the starting cell
+    cell_heuristics[start_coord.row()][start_coord.col()].update(start_coord, 0.0, 0.0);
+    open_list.insert(start_coord);
 
-    bool found_dest = findPathToDestination(dest_coord);
+    bool found_end = findPathToEnd(end_coord);
 
-    if (found_dest == false)
+    if (found_end == false)
     {
         return std::nullopt;
     }
 
-    auto path_points = tracePath(dest_coord);
+    auto path_points = tracePath(end_coord);
 
-    // replace destination with actual destination
+    // replace end with actual end
```

Comment:
Without context, this comment sounds strange.

Reply:
clarified it a bit